simularium/simularium-website
# Feature/share time url

Problem
=======
Users would like to share trajectories at particular timestamps, so we are storing and accessing those timestamps with URL params. 

Closes:
[#418](https://github.com/simularium/simularium-website/issues/418) and [#410 ](https://github.com/simularium/simularium-website/issues/410)

with @meganrm 

Solution
========
This PR should address:
1) the ability to preserve a timepoint in a simulation via a URL param
2) the ability to share a URL with such a timestamp
3) and lay the groundwork/utilities for addressing future URL params (camera position, etc.)


## Type of change

* New feature (non-breaking change which adds functionality)


Change summary:
---------------
Added UI components and functions for handling user input, Redux logics for setting trajectory state via incoming params, and utilities for checking, clearing and resetting URL params.

Steps to Verify:
----------------
1. Verify that the Share button in the top right is disabled when no trajectory is loaded.
2. Load a trajectory, select a time stamp, use  the share button.
3. Try toggling back and forth from default share time to custom.
4. Copy provided URL string into browser and verify that trajectory loads at specified time.
